tests/python/pants_test/base/test_hash_utils.py

```diff
@@ -82,3 +86,83 @@ def check_bad_spec(spec):
     check_bad_spec('/')
     check_bad_spec('foo/1')
     check_bad_spec('1/foo')
+
+
+class CoercingJsonEncodingTest(unittest.TestCase):
+  def _coercing_json_encode(self, o, digest=None):
+    return json.dumps(o, cls=CoercingEncoder)
+
+  def test_normal_object_encoding(self):
+    self.assertEqual(self._coercing_json_encode({}), '{}')
+    self.assertEqual(self._coercing_json_encode(()), '[]')
+    self.assertEqual(self._coercing_json_encode([]), '[]')
+    self.assertEqual(self._coercing_json_encode(set([])), '[]')
+    self.assertEqual(self._coercing_json_encode([{}]), '[{}]')
+    self.assertEqual(self._coercing_json_encode([('a', 3)]), '[["a", 3]]')
+    self.assertEqual(self._coercing_json_encode({'a': 3}), '{"a": 3}')
+    self.assertEqual(self._coercing_json_encode([{'a': 3}]), '[{"a": 3}]')
+    self.assertEqual(self._coercing_json_encode(set([1])), '[1]')
+
+  def test_rejects_ordered_dict(self):
+    with self.assertRaisesRegexp(TypeError, r'CoercingEncoder does not support OrderedDict inputs'):
+      self._coercing_json_encode(OrderedDict([('a', 3)]))
+
+  def test_non_string_dict_key_coercion(self):
+    self.assertEqual(self._coercing_json_encode({('a', 'b'): 'asdf'}),
+                     '{"[\\"a\\", \\"b\\"]": "asdf"}')
+
+  def test_string_like_dict_key_coercion(self):
+    self.assertEqual(self._coercing_json_encode({'a': 3}), '{"a": 3}')
+    self.assertEqual(self._coercing_json_encode({b'a': 3}), '{"a": 3}')
+
+  def test_nested_dict_key_coercion(self):
+    self.assertEqual(self._coercing_json_encode({(1,): {(2,): 3}}),
+                     '{"[1]": {"[2]": 3}}')
+
+  def test_collection_ordering(self):
+    self.assertEqual(self._coercing_json_encode(set([2, 1, 3])), '[1, 2, 3]')
+    self.assertEqual(self._coercing_json_encode({'b': 4, 'a': 3}), '{"a": 3, "b": 4}')
+    self.assertEqual(self._coercing_json_encode([('b', 4), ('a', 3)]), '[["b", 4], ["a", 3]]')
+    self.assertEqual(self._coercing_json_encode([{'b': 4, 'a': 3}]),
+                     '[{"b": 4, "a": 3}]' if PY3 else '[{"a": 3, "b": 4}]')
+
+
+class JsonHashingTest(unittest.TestCase):
+
+  def test_known_checksums(self):
+    """Check a laundry list of supported inputs to stable_json_sha1().
+
+    This checks both that the method can successfully handle the type of input object, but also that
+    the hash of specific objects remains stable.
+    """
+    self.assertEqual(stable_json_sha1({}), 'bf21a9e8fbc5a3846fb05b4fa0859e0917b2202f')
+    self.assertEqual(stable_json_sha1(()), '97d170e1550eee4afc0af065b78cda302a97674c')
+    self.assertEqual(stable_json_sha1([]), '97d170e1550eee4afc0af065b78cda302a97674c')
+    self.assertEqual(stable_json_sha1(set([])), '97d170e1550eee4afc0af065b78cda302a97674c')
```

Comment:
`set([]) == set()` is `True`

```suggestion
    self.assertEqual(stable_json_sha1(set()), '97d170e1550eee4afc0af065b78cda302a97674c')
```